crates/wasi/src/p2/filesystem.rs

```diff
@@ -0,0 +1,573 @@
+use crate::p2::bindings::filesystem::types;
```

Comment:
It looks like this got tracked by git as a delete and re-creation, can you change that into a rename so that we retain history prior to the rename?

Reply:
You're right @rvolosatovs that with the squash-and-merge strategy we have the history here won't get preserved even if the PR preserves the history. That being said could you try renaming `src/filesystem.rs` to something like `src/fs.rs`? That might tip heuristics the other direction to see the rename here. I think the issue is that some of `src/filesystem.rs` was retained while most was moved here, and by having the original copy still preserved it might be confusing the rename detection.

We could then follow-up with a commit to rename `src/fs.rs` to `src/filesystem.rs` if we really wanted too which I think would fixup the git history here.